server/authentication.js

```diff
@@ -8,42 +13,220 @@ function enforceHTTPS(request, response, next) {
   return next();
 }
 
-// Only allow folks who are authorized.  Call `next` if they are authorized,
-// and return a 404 with no body if not.
-function onlyAllowResearchers(request, response, next) {
-  // TODO
-  // 1. Read the token from the request header
-  // 2. Check to see if the token header is in `tokens` database table and it's less than 24 hours old
+function getDomain(request) {
+  return (process.env.NODE_ENV === 'development')
+    ? 'http://localhost:3000'
+    : `https://${request.headers.host}`;
+}
+
+// Middleman function to confirm authorization token is valid
+// Reads token from request header and checks against tokens in db
+function onlyAllowResearchers(pool, request, response, next) {
+  //TODO: Is this really true?
   // 3. This code should run in both development and production.
-  if (process.env.NODE_ENV === 'development') return next();
-  response.status(404);
-  response.end();
+
+  const token = request.headers['token'];
+  const email = request.headers['email'];
+
+  checkToken(pool, email, token)
+    .then(tokenAuthorized => {
```

Comment:
A helpful convention to signal that this is a boolean value is a name like `isTokenAuthorized`.

Reply:
makes sense. done!